### DISCUSSION TITLE
Help with disabling lsp keymaps, and configuring lua_ls

### DISCUSSION THREAD

**Discussion by danielzdev** [NvChad/NvChad#3155] [Q&A]
So, I am fairly new to neovim, so I much prefer a distribution hence why I am using nvchad, its the most beautiful one, in my opinion! So just wanted to give props to the creator :D

Second, I am having trouble disabling keymaps for the lsp stuff, cause I want to create my own, I basically want to disable all of them, or at least close to it. As an example, I use <leader>w to save files, just a thing that I personally like, and now there is always a delay due to all the lsp keymaps with w as prefix.

I also wanted to know, how exactly do I configure lua_ls, as an example, if I wanted to use snacks.nvim, and include Snacks as a global.

This is all from a fresh config by the way, since this is one of the first things I wanted to do , and from there I have kept getting stuck trying everything.

**Comment by siduck** [comment]
you can just not use our on_attach if u want to disable all of them.

if just one two mappings then use vim.keymap.del

but u need to make a custom attach func
```lua
local myattach = function(client, bufnr)
 nvlsp.on_attach(client, bufnr)
 
 -- now add your code here!!!!!!
 
 vim.keymap.del... ( dont forget to use its buf field or else it wont work cuz lsp mappings are buffer scoped )

**Reply by siduck** [reply]
![image](https://github.com/user-attachments/assets/a8654f35-c2f4-4494-bb3b-47abbe14b3eb)

just configure lua_ls normally?
